### PR TITLE
ML Section 4.5 - Weight Decay. 

### PR DESCRIPTION
**Pull request by aksrajvanshi**
Addition: Added the MLP section 4.5 - Weight Decay

Note. - I don't know why I'm having the mlp notebook in the commit. I did fetch the latest changes from upstream master.

### CODE REVIEW

File: jupyter/d2l-java/chapter_mlp/weight-decay.ipynb [hunk 1]
```diff
@@ -0,0 +1,784 @@
...
+    "    float lr = 0.003f;\n",
+    "    \n",
+    "    trainLoss = new float[(num_epochs/5)];\n",
+    "    testLoss = new float[(num_epochs/5)];\n",
+    "    epochCount = new float[(num_epochs/5)];\n",
+    "    \n",
+    "     LearningRateTracker lrt = LearningRateTracker.fixedLearningRate(lr);\n",
+    "     Optimizer sgd = Optimizer.sgd().setLearningRateTracker(lrt).build();\n",
+    "    \n",
+    "     DefaultTrainingConfig config = new DefaultTrainingConfig(l2loss)\n",
```
Comment (zachgk) [inline jupyter/d2l-java/chapter_mlp/weight-decay.ipynb:628]: Fix indentation here

File: jupyter/d2l-java/chapter_mlp/weight-decay.ipynb [hunk 2]
```diff
@@ -0,0 +1,784 @@
...
+    "    epochCount = new float[(num_epochs/5)];\n",
+    "\n",
+    "    for(int epoch = 1; epoch <= num_epochs; epoch++){\n",
+    "        \n",
+    "        for(Batch batch : train_iter.getData(manager)){\n",
+    "            \n",
+    "            NDArray X = batch.getData().head();\n",
+    "            NDArray y = batch.getLabels().head();\n",
+    "            \n",
+    "            // Attach Gradients\n",
```
Comment (zachgk) [inline jupyter/d2l-java/chapter_mlp/weight-decay.ipynb:456]: Try to have the comment indentation match the code indentation. Otherwise, it is hard to read
